--- lua-5.1.4/src/Makefile	2008-01-19 20:37:58.000000000 +0100
+++ lua-5.1.4-patched/src/Makefile	2009-09-30 12:45:46.000000000 +0200
@@ -96,7 +96,7 @@
 	$(MAKE) all MYCFLAGS=
 
 linux:
-	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-Wl,-E -ldl -lreadline -lhistory -lncurses"
+	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-Wl,-E -ldl"
 
 macosx:
 	$(MAKE) all MYCFLAGS=-DLUA_USE_LINUX MYLIBS="-lreadline"
