--- uClibc/utils/Makefile.in	2010-01-13 22:22:31.000000000 +0100
+++ uClibc-working/utils/Makefile.in	2010-01-16 17:35:33.000000000 +0100
@@ -16,7 +16,8 @@
     -I$(top_srcdir)/$(KERNEL_HEADERS) \
     -DNOT_IN_libc \
     -B$(top_builddir)lib \
-    -Wl,-rpath-link,$(top_builddir)lib
+    -Wl,-rpath-link,$(top_builddir)lib \
+    -L$(top_builddir)lib
 
 ifeq ($(UCLIBC_BUILD_PIE),y)
 CFLAGS-utils-shared := $(PIEFLAG) $(LDPIEFLAG)
